testplan/runnable/interactive/base.py

```diff
@@ -157,10 +157,8 @@ def run_all_tests(self, runner_uid=None, await_results=True):
 
         all_tests = self.all_tests(runner_uid)
 
-        return self._create_testplan_report(
+        for test_uid, real_runner_uid in all_tests:
             self.run_test(test_uid, real_runner_uid)
-            for test_uid, real_runner_uid in all_tests
-        )
 
     def run_test(self, test_uid, runner_uid=None, await_results=True):
```

Comment:
What's the reason for changing this to return either `Future<None>` or `None` rather than `Future<testplan.report.TestReport>` or `testplan.report.TestReport`?

Seems that if you returned the report object from `_merge_testcase_reports` and then returned its result in the `with` statement you could get the same signature as the old method with little effort.

Also I'm not jazzed on this one function having two possible return types but that's what the previous function has and I have seen it in the wild (but in *javascript* code) so I'm not horribly fussed about that.

Reply:
Yeah so previously my idea was to magically combine the test reports for only the tests that ran and returning them. But in the end I decided this wasn't really worth the effort or additional computation required to construct such reports, which most of the time will be ignored (at least when running with the web UI driving the API).

Reply:
oh and having an optional return type is pretty common, in a strongly typed language you would say the return type is `Optional<Future<None>>`. In fact, python3 typing has an Optional, so this would directly translate to `Optional[Future]` (not sure if the Future object takes a template param for typing purposes).

Reply:
There's no reason that I can see to break the API. The old method never returns None.

Reply:
This is all relatively new code (this method added two months ago) and its usage throughout testplan is confined only to the interactive runner, which no other part of testplan depends on and which no client is using. So, my concerns about API breakage are moot.